### ISSUE TITLE
[Freature Request] `remap` task support `exclude`

### ISSUE THREAD

**Issue by ExerciseBook** [architectury/architectury-loom#124]
# Reauested Feature
Exclude some class from remapping

# Usage
```kotlin
    remapJar {
        archiveClassifier.set("remap")
        dependsOn(shadowJar)
        mustRunAfter(shadowJar)
        inputFile.set(shadowJar.get().archiveFile.get())
        exclude {
            it.relativePath.contains("relocated packages by shadowJar")
        }
    }
```

# Original Scene

1. `Jar` task pack a jar file (A).
2. `ShadowJar` task add some third party package to the jar file (A) and save to jar file (B).
3. Loom `remap` task try to remap jar file (B), but exception occurred.

![image](https://user-images.githubusercontent.com/6327311/227727084-23d6d185-af1b-493e-b145-bae9b3665a62.png)

After reason locating, I found that the exception was occur when remapping third party class which does not appear in file A but appears in file B

![image](https://user-images.githubusercontent.com/6327311/227727182-0a41e303-4395-496f-962d-2e0552a26d3d.png)

After that, I try to add exclude properity in `remap` task, but it did not work.

I also look up <https://fabricmc.net/wiki/documentation:fabric_loom>, it did not mention `exclude`. 

I think loom may not support `remap exclude` yet, so I create this issue.

cc: @warmthdawn @nicognaW

**Comment by shedaniel** [comment]
You should shadow after remap though... also this is architectury loom, we don't really add these features except those related to architectury or for forge.

**Comment by ExerciseBook** [comment]
Thanks for your reply. I fixed this problem by setting shadow after remap.

I got that there is no plan to add this feature yet, so I close this issue first.

Thank you so much.

**Comment by ExerciseBook** [comment]
Hello, I have another problem now.

When I set shadowjar after remap. I try to set remap's output as shadowjar's input by 
```
shadowJar {
    from(remap)
    // also tried 
    // from(remapJar.get())
    // from(remapJar.get().archiveFile)
    // from(remapJar.get().archiveFile.get())
    // form(remapJar.get().archiveFile.get().asFile)
}
```

but when I unzip shadow's output file, I found that it was not remapped file. So I wonder how I can get remap's output and set it to shadowjar's input.




![image](https://user-images.githubusercontent.com/6327311/227819390-28c0efd3-a9b9-4174-a27a-8e50bd3a4e6b.png)



file from shadow jar
![image](https://user-images.githubusercontent.com/6327311/227819455-874df94e-88b7-4df3-afa0-a6d48d4e7f54.png)

file from remap
![image](https://user-images.githubusercontent.com/6327311/227819494-2c2933a8-2a5e-4793-bf2f-c305f260fd1c.png)
